FSharpLu.Json/Compact.fs

```diff
@@ -204,35 +212,57 @@ type CompactUnionJsonConverter(?tupleAsHeterogneousArray:bool) =
                 FSharpValue.MakeUnion(caseSome, [| nestedValue |])
 
         // Tuple type?
-        else if tupleAsHeterogneousArray && isTupleType objectType then
-            if reader.TokenType <> JsonToken.StartArray then
-                failwithf "Expecting a JSON array, got something else"
+        else if isTupleType objectType then
+            if reader.TokenType = JsonToken.StartArray then
```

Comment:
We should add the condition `&& tupleAsHeterogneousArray` so that we preserve the behavior of the legacy serialization that fails to deserialize arrays into tuples.